File: gems/decomplex/examples/zig/state-branch-density.zig
const StateBranchChecker = struct {
    checked: bool,

    pub fn check(self: *StateBranchChecker, admin: bool, name: []const u8) void {
        if (admin) {
            self.checked = true;
        }

        if (self.checked and name == "admin") {
            print("hello");
        }
    }
};

File: gems/decomplex/examples/rust/state-branch-density.rs
struct StateBranchChecker {
    checked: bool,
}

impl StateBranchChecker {
    fn check(&mut self, admin: bool, name: String) {
        if admin {
            self.checked = true;
        }

        if self.checked && name == "admin" {
            print("hello");
        }
    }
}
